src/lib/swagger.ts

```diff
@@ -2,23 +2,67 @@ import { Application } from 'express';
 import * as swaggerUi from 'swagger-ui-express';
 import * as yaml from 'js-yaml';
 import * as fs from 'fs';
+import * as path from 'path';
 
 /**
  * Serve swagger documentation
  */
 export function setSwagger(app: Application, route: string, filePath: string, options: SwaggerOptions = {}): void {
   try {
-    // Load yaml document
-    const swaggerDocument = yaml.safeLoad(fs.readFileSync(filePath, 'utf8'));
-    Object.assign(swaggerDocument, options);
+    const stats = fs.lstatSync(filePath);
```

Comment:
if (options.concatenate) {
      // Throw error if concatenate = true and filepath = file
      if (stats.isFile()) {
        throw new Error('Boolean concatenate cannot be true when you specify a file. When you want to concatenate, specify a folder');
      }
      if (stats.isDirectory()) {
        const swaggerContent = buildSwaggerDocumentFromFiles(filePath);
        swaggerDocument = yaml.safeLoad(swaggerContent);
      }
    } else {
      if (stats.isFile()) {
        // Load yaml document
        swaggerDocument = yaml.safeLoad(fs.readFileSync(filePath, 'utf8'));
        Object.assign(swaggerDocument, options);
      }
      if (stats.isDirectory()) {
        throw new Error('To concatenate a folder of swagger YMLS, you need to explicitly set the boolean concatenate on true for the swaggerOptions');
      }
    }

Yes because i throw an Error when the file is a directory and concatenate is not set